pkg/cnab/provider/action.go

```diff
@@ -52,3 +60,23 @@ func (d *Runtime) AddFiles(args ActionArguments) action.OperationConfigFunc {
 		return nil
 	}
 }
+
+// AddRelocation operates on an ActionArguments and adds any provided relocation mapping
+// to the operation's files.
+func (d *Runtime) AddRelocation(args ActionArguments) action.OperationConfigFunc {
+	return func(op *driver.Operation) error {
+		if args.RelocationMapping != "" {
+			b, err := d.FileSystem.ReadFile(args.RelocationMapping)
+			if err != nil {
+				return errors.Wrap(err, "unable to add relocation mapping")
+			}
+			op.Files["/cnab/app/relocation-mapping.json"] = string(b)
+			var reloMap relocation.ImageRelocationMap
+			err = json.Unmarshal(b, &reloMap)
+			if mappedInvo, ok := reloMap[op.Image.Image]; ok {
```

Comment:
Here we update the operation invocation image based on the value presented in the relo map, if exists, right?